src/Chapter4.hs

```diff
@@ -673,6 +690,19 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data BTree a = Leaf | Node a (BTree a) (BTree a)
+
+instance Functor BTree where
+  fmap _ Leaf = Leaf
+  fmap f (Node v l r) = Node (f v) (fmap f l) (fmap f r)
+
+reverseTree :: BTree a -> BTree a
+reverseTree Leaf = Leaf
+reverseTree (Node v l r) = Node v (reverseTree r) (reverseTree l)
+
+toList :: BTree a -> List a
+toList Leaf = Empty
+toList (Node v l r) = Cons v $ conc $ Cons (toList l) $ Cons (toList r) Empty
```

Comment:
Interesting!
Usually, people submit the solution for `[]`, but this one is correct, nice job 👍🏼